auth-tokens/src/main/java/com/palantir/tokens/auth/BearerToken.java

```diff
@@ -49,4 +58,8 @@ public static BearerToken valueOf(String token) {
     public final String toString() {
         return getToken();
     }
+
+    public final JsonWebToken asJwt() {
```

Comment:
It could be, asJwt will throw if it's not. There's some more discussion of this approach here: https://github.com/palantir/auth-tokens/pull/9

If you prefer the getter on BearerToken we can go with #9 instead, but I think this makes it a bit clearer what the setup is and gives us the opportunity to reveal other encoded/untrusted data (as well as provide some options for indicating trust or offering to verify things). 

I did add the optional fields in for the payload parser, we could also expose those but I couldn't decide on how to differentiate between session and API tokens reasonably.